test/test_buildpacks.go

```diff
@@ -0,0 +1,207 @@
+///usr/bin/true; exec /usr/bin/env go run "$0" "$@"
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+const repo = "ghcr.io/boson-project"
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		<-sigs
+		cancel()
+		<-sigs
+		os.Exit(1)
+	}()
+
+	version := os.Args[1]
+	err := runTests(ctx, version)
+	if err != nil {
+		fmt.Printf("::error::%s\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+var buildpacks = []struct {
+	Buildpacks []string
+	Runtimes   []string
+}{
+	{
+		Buildpacks: []string{"ghcr.io/boson-project/go-function-buildpack", "paketo-buildpacks/go-dist"},
+		Runtimes:   []string{"go"},
+	},
+	{
+		Buildpacks: []string{"ghcr.io/boson-project/typescript-function-buildpack", "paketo-buildpacks/nodejs"},
+		Runtimes:   []string{"typescript"},
+	},
+}
+
+func runTests(ctx context.Context, version string) error {
+
+	os.Setenv("FUNC_REGISTRY", repo)
+
+	for i := range buildpacks {
+		buildpacks[i].Buildpacks[0] = fmt.Sprintf("%s:%s", buildpacks[i].Buildpacks[0], version)
+	}
```

Comment:
This could be replaced with a function which returns a fully populated slice of test cases, removing the need for a global variable and the post-initialization version-append.  Something perhaps like
```
type testCase struct {
    Buildpacks []string
    Runtimes   []string
}
func testCases(version string) []testCase {
    ...
}
```